### PR TITLE
Use appropriate icons for toasts

### PR DESCRIPTION
**Pull request by mozzius**
Despite toasts being used heavily for reporting errors, we just show a default tick icon which is a bit dissonant for a popup displaying bad news. This PR goes through and uses `xmark` for negative ones, with a few as `info` or `exclamation-circle` where appropriate. I also made the clipboard related ones use a clipboard icon.

Also, I shrunk the icon a little and applied `flex-shrink: 0` so it stays a consistent size

Screenshot 2024-07-19 at 14 56 57
Screenshot 2024-07-19 at 14 56 33

### CODE REVIEW

**Comment by haileyok** [comment]
Do we want to do the same for native?

File: src/components/dms/dialogs/NewChatDialog.tsx [hunk 1]
```diff
@@ -31,7 +32,8 @@ export function NewChat({
       logEvent('chat:open', {logContext: 'NewChatDialog'})
     },
     onError: error => {
-      Toast.show(error.message)
+      logger.error('Failed to create chat', {message: error})
```
Comment (haileyok) [inline src/components/dms/dialogs/NewChatDialog.tsx:35]: i think this needs to be `safeMessage`

**Comment by mozzius** [comment]
> Do we want to do the same for native?

https://github.com/bluesky-social/social-app/pull/4808

**Review by haileyok — APPROVED** [review]

File: src/lib/media/picker.shared.ts [hunk 2]
```diff
@@ -17,14 +17,14 @@ export async function openPicker(opts?: ImagePickerOptions) {
   })
 
   if (response.assets && response.assets.length > 4) {
-    Toast.show('You may only select up to 4 images')
+    Toast.show('You may only select up to 4 images', 'exclamation-circle')
```
Comment (surfdude29) [inline src/lib/media/picker.shared.ts:20]: While you're updating toasts, would it be possible to mark this for localization please? 🙏

File: src/lib/media/picker.shared.ts [hunk 3]
```diff
@@ -17,14 +17,14 @@ export async function openPicker(opts?: ImagePickerOptions) {
...
   }
 
   return (response.assets ?? [])
     .slice(0, 4)
     .filter(asset => {
       if (asset.mimeType?.startsWith('image/')) return true
-      Toast.show('Only image files are supported')
+      Toast.show('Only image files are supported', 'exclamation-circle')
```
Comment (surfdude29) [inline src/lib/media/picker.shared.ts:27]: And here

File: src/lib/sharing.ts [hunk 4]
```diff
@@ -20,6 +20,6 @@ export async function shareUrl(url: string) {
     // React Native Share is not supported by web. Web Share API
     // has increasing but not full support, so default to clipboard
     setStringAsync(url)
-    Toast.show('Copied to clipboard')
+    Toast.show('Copied to clipboard', 'clipboard-check')
```
Comment (surfdude29) [inline src/lib/sharing.ts:23]: Also here

File: src/components/dms/dialogs/NewChatDialog.tsx (hunk 1 above)
Comment (mozzius) [reply src/components/dms/dialogs/NewChatDialog.tsx:35]: Seems like it's super inconsistent across the whole codebase, I'll change it here but we should look at it in other places too

**Pull request merged**